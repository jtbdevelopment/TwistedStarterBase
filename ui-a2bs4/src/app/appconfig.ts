import {Injectable} from '@angular/core';
//  TODO - TSB fix name here
@Injectable()
export class AppConfig {
    appName: string = 'Twisted Starter Base';
}

