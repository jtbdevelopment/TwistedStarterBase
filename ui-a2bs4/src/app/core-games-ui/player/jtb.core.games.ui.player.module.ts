import {NgModule} from '@angular/core';
import {HttpModule} from '@angular/http';
import {JTBCoreGamesUIMessageBus} from '../messagebus/jtb.core.games.ui.messagebus.module';
import {PlayerService} from './player.service';

@NgModule({
    imports: [HttpModule, JTBCoreGamesUIMessageBus],
    exports: [JTBCoreGamesUIMessageBus],
    providers: [
        PlayerService
    ]
})
export class JTBCoreGamesUIPlayer {
}
