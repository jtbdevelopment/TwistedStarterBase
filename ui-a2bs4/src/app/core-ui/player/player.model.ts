
export class Player {
    id: string;
    md5: string;
    source: string;
    displayName: string;
    imageUrl: string;
    profileUrl: string;
    lastVersionNotes: string;
    adminUser: boolean = false;
    gameSpecificPlayerAttributes: any = {};

    constructor(copyFrom?: Player) {
        if (copyFrom) {
            this.id = copyFrom.id;
            this.md5 = copyFrom.md5;
            this.source = copyFrom.source;
            this.displayName = copyFrom.displayName;
            this.imageUrl = copyFrom.imageUrl;
            this.profileUrl = copyFrom.profileUrl;
            this.lastVersionNotes = copyFrom.lastVersionNotes;
            this.adminUser = copyFrom.adminUser;
            this.gameSpecificPlayerAttributes = Object.assign({}, copyFrom.gameSpecificPlayerAttributes);
        }
    }
}
