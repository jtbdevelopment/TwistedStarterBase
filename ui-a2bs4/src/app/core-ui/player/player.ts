
export class Player {
    source: string;
    displayName: string;
    imageUrl: string;
    profileUrl: string;
    lastVersionNotes: string;
    adminUser: boolean;
    //  TODO
    //gameSpecificPlayerAttributes: Map<string, string> = new Map<string, string>();
}
