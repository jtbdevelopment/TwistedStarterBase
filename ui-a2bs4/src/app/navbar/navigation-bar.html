<!--  TODO - break down to further components?  -->
<nav class="navbar navbar-toggleable-sm navbar-light bg-faded fixed-bottom">
    <navigation-bar-game-menu-toggle [playerLoaded]="playerLoaded"></navigation-bar-game-menu-toggle>
    <navigation-bar-new-game [playerLoaded]="playerLoaded"></navigation-bar-new-game>
    <a class="navbar-brand" href="/">{{appName}}
        <small>Welcome {{playerName}}</small>
    </a>
    <ul class="navbar-nav ml-auto float-right">
        <li>
            <!--  TODO -->
            <a routerLink="/profile" *ngIf="playerLoaded" class="nav-link">
                <span class="fa fa-user"></span>
            </a>
        </li>
        <li>
            <!--  TODO -->
            <a routerLink="/help" *ngIf="playerLoaded" class="nav-link">
                <span class="fa fa-question"></span>
            </a>
        </li>
        <li>
            <!--  TODO -->
            <a routerLink="/admin" *ngIf="showAdmin" class="nav-link">
                <span class="fa fa-eye"></span>
            </a>
        </li>
        <!--  TODO -->
        <li><a href="javascript:void(0)" (click)="logout()" *ngIf="showLogout" class="nav-link">Logout</a></li>
    </ul>
</nav>
