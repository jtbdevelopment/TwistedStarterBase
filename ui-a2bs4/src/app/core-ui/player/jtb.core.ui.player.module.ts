import {NgModule} from '@angular/core';
import {HttpModule} from '@angular/http';
import {JTBCoreUIMessageBus} from '../messagebus/jtb.core.ui.messagebus.module';
import {PlayerService} from './player.service';

@NgModule({
    imports: [HttpModule, JTBCoreUIMessageBus],
    exports: [JTBCoreUIMessageBus],
    providers: [
        PlayerService
    ]
})
export class JTBCoreUIPlayer {
}
