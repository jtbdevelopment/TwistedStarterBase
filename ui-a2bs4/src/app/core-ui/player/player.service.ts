import {Injectable} from '@angular/core';
import {Player} from './player.model';
import {Observable, BehaviorSubject} from 'rxjs';
import {Http} from '@angular/http';
import {MessageBusService} from '../messagebus/message-bus.service';
import {Router} from '@angular/router';

@Injectable()
export class PlayerService {
    //  Unless dealing in admin functions use player
    //  In general, these two will be the same unless an admin player
    //  switches to simulation mode - then loggedInPlayer will retain the admin details
    //  while player will switch to the simulated player
    player: Observable<Player>;
    loggedInPlayer: Observable<Player>;

    private playerSubject: BehaviorSubject<Player> = new BehaviorSubject(new Player());
    private loggedInSubject: BehaviorSubject<Player> = new BehaviorSubject(new Player());

    //  TODO - not currently possible to test message bus updates when logged in != current
    constructor(private http: Http,
                private messageBus: MessageBusService,
                private router: Router) {
        this.player = Observable.from<Player>(this.playerSubject);
        this.loggedInPlayer = Observable.from<Player>(this.loggedInSubject);
        this.messageBus.playerUpdates.subscribe(player => {
            if (player.id === this.playerSubject.getValue().id) {
                console.log('player update ' + JSON.stringify(player));
                this.playerSubject.next(player);
            }
            if (player.id === this.loggedInSubject.getValue().id) {
                console.log('logged in update ' + JSON.stringify(player));
                this.loggedInSubject.next(player);
            }
        });
    }

    public loadLoggedInPlayer(): void {
        this.http.get('/api/security')
            .map(response => response.json())
            .map(json => {
                let loaded = new Player(json);
                return loaded;
            })
            .subscribe(loaded => {
                    this.playerSubject.next(loaded);
                    this.loggedInSubject.next(loaded);
                },
                error => {
                    //  TODO - general error handler
                    console.log(JSON.stringify(error));
                });
    }

    public logout(): void {
        this.http.post('/signout', null).subscribe(() => {
            this.resetPlayer();
        }, error => {
            console.log(JSON.stringify(error));
            this.resetPlayer();
        });
    }

    private resetPlayer(): void {
        this.loggedInSubject.next(new Player());
        this.playerSubject.next(new Player());
        this.messageBus.playerUpdates.next(new Player());
        this.router.navigateByUrl('/signin');
    }
}
