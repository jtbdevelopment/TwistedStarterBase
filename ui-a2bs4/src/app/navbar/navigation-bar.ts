import {Component, NgModule} from '@angular/core';
import {AppConfig} from '../appconfig';
import {PlayerService} from '../core-ui/player/playerService';
import {BrowserModule} from '@angular/platform-browser';
import {JTBCoreUI} from '../core-ui/jtb.core.ui';
import {NavigationBarGameMenuToggleComponent} from './navigation-bar-game-menu-toggle';
import {NavigationBarNewGameComponent} from './navigation-bar-new-game';
import {NavigationBarRightMenuComponent} from './navigation-bar-right-menu';

@Component({
    selector: 'navigation-bar',
    template: require('./navigation-bar.html'),
    styles: [require('./navigation-bar.scss').toString()]
})
export class NavigationBarComponent {
    playerName: string;
    showAdmin: boolean = false;
    showLogout: boolean = false;
    appName: string;
    playerLoaded: boolean = false;

    constructor(config: AppConfig, playerService: PlayerService) {
        this.appName = config.appName;
        playerService.loggedInPlayer.subscribe(login => {
            this.showAdmin = login.adminUser;
            this.playerLoaded =
                login !== undefined &&
                login.displayName !== undefined &&
                login.displayName !== '';
        });
        playerService.player.subscribe(user => {
            this.showLogout = (user.source === 'Manual');
            this.playerName = user.displayName;
        });
    }

    logout(): void {
        //  TODO
        console.log('logout ' + this.playerName);
    }
}

@NgModule({
    imports: [
        BrowserModule,
        JTBCoreUI,
    ],
    exports: [
        NavigationBarComponent
    ],
    declarations: [
        NavigationBarComponent,
        NavigationBarGameMenuToggleComponent,
        NavigationBarNewGameComponent,
        NavigationBarRightMenuComponent
    ]
})
export class NavigationBarModule {
}

