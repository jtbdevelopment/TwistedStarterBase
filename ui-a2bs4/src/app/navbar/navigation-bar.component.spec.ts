import {async, TestBed} from '@angular/core/testing';
import {NavigationBarComponent} from './navigation-bar.component';
import {Component, Input} from '@angular/core';
import {PlayerService} from '../core-games-ui/player/player.service';
import {Player} from '../core-games-ui/player/player.model';
import {BehaviorSubject, Observable} from 'rxjs';
import {AppConfig} from '../core-games-ui/appconfig.interface';

@Component({
    selector: 'navigation-bar-game-menu-toggle',
    template: '<p>{{playerLoaded}}</p>'
})
export class MockGameMenuToggleComponent {
    @Input() playerLoaded: boolean;
}

@Component({
    selector: 'navigation-bar-new-game',
    template: '<p>{{playerLoaded}}</p>'
})
export class MockNewGameComponent {
    @Input() playerLoaded: boolean;
}

@Component({
    selector: 'navigation-bar-right-menu',
    template: '<p>{{playerLoaded}}/{{showAdmin}}/{{showLogout}}</p>'
})
export class MockRightBarMenuComponent {
    @Input() playerLoaded: boolean;
    @Input() showAdmin: boolean;
    @Input() showLogout: boolean;
}

export class MockAppConfig implements AppConfig {
    appName: string = 'Test';
    hoverMenu: boolean;
    version: string = '';
    releaseNotes: string = '';
}

export class MockPlayerService {
    static playerSubject: BehaviorSubject<Player> = new BehaviorSubject(new Player());
    static loggedInSubject: BehaviorSubject<Player> = new BehaviorSubject(new Player());

    player: Observable<Player> = Observable.from<Player>(MockPlayerService.playerSubject);
    loggedInPlayer: Observable<Player> = Observable.from<Player>(MockPlayerService.loggedInSubject);
}

describe('nav bar component', () => {
    beforeEach(async(() => {
        TestBed.configureTestingModule({
            declarations: [
                NavigationBarComponent,
                MockGameMenuToggleComponent,
                MockRightBarMenuComponent,
                MockNewGameComponent
            ],
            providers: [
                {provide: 'AppConfig', useClass: MockAppConfig},
                {provide: PlayerService, useClass: MockPlayerService}
            ],
        });
        TestBed.compileComponents();
        MockPlayerService.loggedInSubject.next(new Player());
        MockPlayerService.playerSubject.next(new Player());
    }));

    it('should render basics when no player logged in', () => {
        const fixture = TestBed.createComponent(NavigationBarComponent);
        fixture.detectChanges();
        const navBar = fixture.nativeElement;
        expect(fixture.componentInstance.playerName).toBeUndefined();
        expect(fixture.componentInstance.playerLoaded).toBeFalsy();
        expect(fixture.componentInstance.showLogout).toBeFalsy();
        expect(fixture.componentInstance.showAdmin).toBeFalsy();
        expect(fixture.componentInstance.appName).toEqual('Test');
        expect(navBar.querySelector('navigation-bar-game-menu-toggle').textContent.trim()).toBe('false');
        expect(navBar.querySelector('navigation-bar-new-game').textContent.trim()).toBe('false');
        expect(navBar.querySelector('navigation-bar-right-menu').textContent.trim()).toBe('false/false/false');
        expect(navBar.querySelector('a').textContent.trim()).toBe('Test\n        Welcome');
    });

    it('should update for a regular non manual player', () => {
        const fixture = TestBed.createComponent(NavigationBarComponent);
        fixture.detectChanges();
        const navBar = fixture.nativeElement;
        expect(fixture.componentInstance.playerName).toBeUndefined();
        expect(fixture.componentInstance.playerLoaded).toBeFalsy();
        expect(fixture.componentInstance.showLogout).toBeFalsy();
        expect(fixture.componentInstance.showAdmin).toBeFalsy();
        expect(fixture.componentInstance.appName).toEqual('Test');
        expect(navBar.querySelector('navigation-bar-game-menu-toggle').textContent.trim()).toBe('false');
        expect(navBar.querySelector('navigation-bar-new-game').textContent.trim()).toBe('false');
        expect(navBar.querySelector('navigation-bar-right-menu').textContent.trim()).toBe('false/false/false');
        expect(navBar.querySelector('a').textContent.trim()).toBe('Test\n        Welcome');

        let p: Player = new Player();
        p.displayName = 'XYZ';
        p.source = 'Facebook';
        p.adminUser = false;

        MockPlayerService.loggedInSubject.next(p);
        MockPlayerService.playerSubject.next(p);
        fixture.detectChanges();

        expect(fixture.componentInstance.playerName).toEqual(p.displayName);
        expect(fixture.componentInstance.playerLoaded).toBeTruthy();
        expect(fixture.componentInstance.showLogout).toBeFalsy();
        expect(fixture.componentInstance.showAdmin).toBeFalsy();
        expect(fixture.componentInstance.appName).toEqual('Test');
        expect(navBar.querySelector('navigation-bar-game-menu-toggle').textContent.trim()).toBe('true');
        expect(navBar.querySelector('navigation-bar-new-game').textContent.trim()).toBe('true');
        expect(navBar.querySelector('navigation-bar-right-menu').textContent.trim()).toBe('true/false/false');
        expect(navBar.querySelector('a').textContent.trim()).toBe('Test\n        Welcome XYZ');
    });

    it('should update for a regular manual player', () => {
        const fixture = TestBed.createComponent(NavigationBarComponent);
        fixture.detectChanges();
        const navBar = fixture.nativeElement;
        expect(fixture.componentInstance.playerName).toBeUndefined();
        expect(fixture.componentInstance.playerLoaded).toBeFalsy();
        expect(fixture.componentInstance.showLogout).toBeFalsy();
        expect(fixture.componentInstance.showAdmin).toBeFalsy();
        expect(fixture.componentInstance.appName).toEqual('Test');
        expect(navBar.querySelector('navigation-bar-game-menu-toggle').textContent.trim()).toBe('false');
        expect(navBar.querySelector('navigation-bar-new-game').textContent.trim()).toBe('false');
        expect(navBar.querySelector('navigation-bar-right-menu').textContent.trim()).toBe('false/false/false');
        expect(navBar.querySelector('a').textContent.trim()).toBe('Test\n        Welcome');

        let p: Player = new Player();
        p.displayName = 'XYZ';
        p.source = 'MANUAL';
        p.adminUser = false;

        MockPlayerService.loggedInSubject.next(p);
        MockPlayerService.playerSubject.next(p);
        fixture.detectChanges();

        expect(fixture.componentInstance.playerName).toEqual(p.displayName);
        expect(fixture.componentInstance.playerLoaded).toBeTruthy();
        expect(fixture.componentInstance.showLogout).toBeTruthy();
        expect(fixture.componentInstance.showAdmin).toBeFalsy();
        expect(fixture.componentInstance.appName).toEqual('Test');
        expect(navBar.querySelector('navigation-bar-game-menu-toggle').textContent.trim()).toBe('true');
        expect(navBar.querySelector('navigation-bar-new-game').textContent.trim()).toBe('true');
        expect(navBar.querySelector('navigation-bar-right-menu').textContent.trim()).toBe('true/false/true');
        expect(navBar.querySelector('a').textContent.trim()).toBe('Test\n        Welcome XYZ');
    });

    it('should update for a admin non manual player', () => {
        const fixture = TestBed.createComponent(NavigationBarComponent);
        fixture.detectChanges();
        const navBar = fixture.nativeElement;
        expect(fixture.componentInstance.playerName).toBeUndefined();
        expect(fixture.componentInstance.playerLoaded).toBeFalsy();
        expect(fixture.componentInstance.showLogout).toBeFalsy();
        expect(fixture.componentInstance.showAdmin).toBeFalsy();
        expect(fixture.componentInstance.appName).toEqual('Test');
        expect(navBar.querySelector('navigation-bar-game-menu-toggle').textContent.trim()).toBe('false');
        expect(navBar.querySelector('navigation-bar-new-game').textContent.trim()).toBe('false');
        expect(navBar.querySelector('navigation-bar-right-menu').textContent.trim()).toBe('false/false/false');
        expect(navBar.querySelector('a').textContent.trim()).toBe('Test\n        Welcome');

        let p: Player = new Player();
        p.displayName = 'XYZ';
        p.source = 'Facebook';
        p.adminUser = true;

        MockPlayerService.loggedInSubject.next(p);
        MockPlayerService.playerSubject.next(p);
        fixture.detectChanges();

        expect(fixture.componentInstance.playerName).toEqual(p.displayName);
        expect(fixture.componentInstance.playerLoaded).toBeTruthy();
        expect(fixture.componentInstance.showLogout).toBeFalsy();
        expect(fixture.componentInstance.showAdmin).toBeTruthy();
        expect(fixture.componentInstance.appName).toEqual('Test');
        expect(navBar.querySelector('navigation-bar-game-menu-toggle').textContent.trim()).toBe('true');
        expect(navBar.querySelector('navigation-bar-new-game').textContent.trim()).toBe('true');
        expect(navBar.querySelector('navigation-bar-right-menu').textContent.trim()).toBe('true/true/false');
        expect(navBar.querySelector('a').textContent.trim()).toBe('Test\n        Welcome XYZ');
    });
});
