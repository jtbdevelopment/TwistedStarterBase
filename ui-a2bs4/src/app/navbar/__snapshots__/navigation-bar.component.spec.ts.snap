// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav bar component should render basics when no player logged in 1`] = `
<app-navigation-bar
  appName={[Function String]}
  config={[Function MockAppConfig]}
  playerLoaded="false"
  playerName="undefined"
  playerService={[Function MockPlayerService]}
  showAdmin="false"
  showLogout="false"
>
  <nav
    class="navbar navbar-toggleable-sm navbar-light fixed-bottom"
  >
    
      
    <app-navigation-bar-game-menu-toggle
      ng-reflect-player-loaded="false"
    >
      <p>
        false
      </p>
    </app-navigation-bar-game-menu-toggle>
    
      
    <app-navigation-bar-new-game
      ng-reflect-player-loaded="false"
    >
      <p>
        false
      </p>
    </app-navigation-bar-new-game>
    
      
    <a
      class="navbar-brand"
      href="/"
    >
      Test
          
      <small>
        Welcome 
      </small>
      
      
    </a>
    
      
    <app-navigation-bar-right-menu
      class="ml-auto float-right"
      ng-reflect-player-loaded="false"
      ng-reflect-show-admin="false"
      ng-reflect-show-logout="false"
    >
      <p>
        false/false/false
      </p>
    </app-navigation-bar-right-menu>
    
  
  </nav>
  
</app-navigation-bar>
`;

exports[`nav bar component should update for a admin non manual player 1`] = `
<app-navigation-bar
  appName={[Function String]}
  config={[Function MockAppConfig]}
  playerLoaded="false"
  playerName="undefined"
  playerService={[Function MockPlayerService]}
  showAdmin="false"
  showLogout="false"
>
  <nav
    class="navbar navbar-toggleable-sm navbar-light fixed-bottom"
  >
    
      
    <app-navigation-bar-game-menu-toggle
      ng-reflect-player-loaded="false"
    >
      <p>
        false
      </p>
    </app-navigation-bar-game-menu-toggle>
    
      
    <app-navigation-bar-new-game
      ng-reflect-player-loaded="false"
    >
      <p>
        false
      </p>
    </app-navigation-bar-new-game>
    
      
    <a
      class="navbar-brand"
      href="/"
    >
      Test
          
      <small>
        Welcome 
      </small>
      
      
    </a>
    
      
    <app-navigation-bar-right-menu
      class="ml-auto float-right"
      ng-reflect-player-loaded="false"
      ng-reflect-show-admin="false"
      ng-reflect-show-logout="false"
    >
      <p>
        false/false/false
      </p>
    </app-navigation-bar-right-menu>
    
  
  </nav>
  
</app-navigation-bar>
`;

exports[`nav bar component should update for a admin non manual player 2`] = `
<app-navigation-bar
  appName={[Function String]}
  config={[Function MockAppConfig]}
  playerLoaded={[Function Boolean]}
  playerName={[Function String]}
  playerService={[Function MockPlayerService]}
  showAdmin={[Function Boolean]}
  showLogout="false"
>
  <nav
    class="navbar navbar-toggleable-sm navbar-light fixed-bottom"
  >
    
      
    <app-navigation-bar-game-menu-toggle
      ng-reflect-player-loaded="true"
    >
      <p>
        true
      </p>
    </app-navigation-bar-game-menu-toggle>
    
      
    <app-navigation-bar-new-game
      ng-reflect-player-loaded="true"
    >
      <p>
        true
      </p>
    </app-navigation-bar-new-game>
    
      
    <a
      class="navbar-brand"
      href="/"
    >
      Test
          
      <small>
        Welcome XYZ
      </small>
      
      
    </a>
    
      
    <app-navigation-bar-right-menu
      class="ml-auto float-right"
      ng-reflect-player-loaded="true"
      ng-reflect-show-admin="true"
      ng-reflect-show-logout="false"
    >
      <p>
        true/true/false
      </p>
    </app-navigation-bar-right-menu>
    
  
  </nav>
  
</app-navigation-bar>
`;

exports[`nav bar component should update for a regular manual player 1`] = `
<app-navigation-bar
  appName={[Function String]}
  config={[Function MockAppConfig]}
  playerLoaded="false"
  playerName="undefined"
  playerService={[Function MockPlayerService]}
  showAdmin="false"
  showLogout="false"
>
  <nav
    class="navbar navbar-toggleable-sm navbar-light fixed-bottom"
  >
    
      
    <app-navigation-bar-game-menu-toggle
      ng-reflect-player-loaded="false"
    >
      <p>
        false
      </p>
    </app-navigation-bar-game-menu-toggle>
    
      
    <app-navigation-bar-new-game
      ng-reflect-player-loaded="false"
    >
      <p>
        false
      </p>
    </app-navigation-bar-new-game>
    
      
    <a
      class="navbar-brand"
      href="/"
    >
      Test
          
      <small>
        Welcome 
      </small>
      
      
    </a>
    
      
    <app-navigation-bar-right-menu
      class="ml-auto float-right"
      ng-reflect-player-loaded="false"
      ng-reflect-show-admin="false"
      ng-reflect-show-logout="false"
    >
      <p>
        false/false/false
      </p>
    </app-navigation-bar-right-menu>
    
  
  </nav>
  
</app-navigation-bar>
`;

exports[`nav bar component should update for a regular manual player 2`] = `
<app-navigation-bar
  appName={[Function String]}
  config={[Function MockAppConfig]}
  playerLoaded={[Function Boolean]}
  playerName={[Function String]}
  playerService={[Function MockPlayerService]}
  showAdmin="false"
  showLogout={[Function Boolean]}
>
  <nav
    class="navbar navbar-toggleable-sm navbar-light fixed-bottom"
  >
    
      
    <app-navigation-bar-game-menu-toggle
      ng-reflect-player-loaded="true"
    >
      <p>
        true
      </p>
    </app-navigation-bar-game-menu-toggle>
    
      
    <app-navigation-bar-new-game
      ng-reflect-player-loaded="true"
    >
      <p>
        true
      </p>
    </app-navigation-bar-new-game>
    
      
    <a
      class="navbar-brand"
      href="/"
    >
      Test
          
      <small>
        Welcome XYZ
      </small>
      
      
    </a>
    
      
    <app-navigation-bar-right-menu
      class="ml-auto float-right"
      ng-reflect-player-loaded="true"
      ng-reflect-show-admin="false"
      ng-reflect-show-logout="true"
    >
      <p>
        true/false/true
      </p>
    </app-navigation-bar-right-menu>
    
  
  </nav>
  
</app-navigation-bar>
`;

exports[`nav bar component should update for a regular non manual player 1`] = `
<app-navigation-bar
  appName={[Function String]}
  config={[Function MockAppConfig]}
  playerLoaded="false"
  playerName="undefined"
  playerService={[Function MockPlayerService]}
  showAdmin="false"
  showLogout="false"
>
  <nav
    class="navbar navbar-toggleable-sm navbar-light fixed-bottom"
  >
    
      
    <app-navigation-bar-game-menu-toggle
      ng-reflect-player-loaded="false"
    >
      <p>
        false
      </p>
    </app-navigation-bar-game-menu-toggle>
    
      
    <app-navigation-bar-new-game
      ng-reflect-player-loaded="false"
    >
      <p>
        false
      </p>
    </app-navigation-bar-new-game>
    
      
    <a
      class="navbar-brand"
      href="/"
    >
      Test
          
      <small>
        Welcome 
      </small>
      
      
    </a>
    
      
    <app-navigation-bar-right-menu
      class="ml-auto float-right"
      ng-reflect-player-loaded="false"
      ng-reflect-show-admin="false"
      ng-reflect-show-logout="false"
    >
      <p>
        false/false/false
      </p>
    </app-navigation-bar-right-menu>
    
  
  </nav>
  
</app-navigation-bar>
`;

exports[`nav bar component should update for a regular non manual player 2`] = `
<app-navigation-bar
  appName={[Function String]}
  config={[Function MockAppConfig]}
  playerLoaded={[Function Boolean]}
  playerName={[Function String]}
  playerService={[Function MockPlayerService]}
  showAdmin="false"
  showLogout="false"
>
  <nav
    class="navbar navbar-toggleable-sm navbar-light fixed-bottom"
  >
    
      
    <app-navigation-bar-game-menu-toggle
      ng-reflect-player-loaded="true"
    >
      <p>
        true
      </p>
    </app-navigation-bar-game-menu-toggle>
    
      
    <app-navigation-bar-new-game
      ng-reflect-player-loaded="true"
    >
      <p>
        true
      </p>
    </app-navigation-bar-new-game>
    
      
    <a
      class="navbar-brand"
      href="/"
    >
      Test
          
      <small>
        Welcome XYZ
      </small>
      
      
    </a>
    
      
    <app-navigation-bar-right-menu
      class="ml-auto float-right"
      ng-reflect-player-loaded="true"
      ng-reflect-show-admin="false"
      ng-reflect-show-logout="false"
    >
      <p>
        true/false/false
      </p>
    </app-navigation-bar-right-menu>
    
  
  </nav>
  
</app-navigation-bar>
`;
